### PR TITLE
Kustomize fixes

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:

Fixes issues with running kustomize against `config/default`

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: config/crd/kustomization.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ resources:
 - bases/infrastructure.cluster.x-k8s.io_dockerclusters.yaml
 # +kubebuilder:scaffold:crdkustomizeresource
 
-patches:
+#patches:
```
Comment (chuckha) [inline config/crd/kustomization.yaml]:
```suggestion
patchesStrategicMerge: []
```

let's update it and keep it around. `patches` is simply invalid in kustomize v3.x

**Comment by detiber** [comment]
@chuckha should be good to go now

**Pull request merged**